x-pack/plugins/observability_solution/apm/server/routes/entities/get_service_entities_history_timeseries.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { termsQuery, rangeQuery } from '@kbn/observability-plugin/server';
+import { getBucketSize } from '@kbn/apm-data-access-plugin/common';
+import { keyBy } from 'lodash';
+import {
+  ENTITY_METRICS_FAILED_TRANSACTION_RATE,
+  ENTITY_METRICS_LATENCY,
+  ENTITY_METRICS_LOG_ERROR_RATE,
+  ENTITY_METRICS_LOG_RATE,
+  ENTITY_METRICS_THROUGHPUT,
+  LAST_SEEN,
+} from '../../../common/es_fields/entities';
+import { SERVICE_NAME } from '../../../common/es_fields/apm';
+import { EntitiesESClient } from '../../lib/helpers/create_es_client/create_assets_es_client/create_assets_es_clients';
+import { environmentQuery } from '../../../common/utils/environment_query';
+
+interface Params {
+  entitiesESClient: EntitiesESClient;
+  start: number;
+  end: number;
+  serviceNames: string[];
+  environment: string;
+}
+
+export async function getServiceEntitiesHistoryTimeseries({
+  start,
+  end,
+  serviceNames,
+  entitiesESClient,
+  environment,
+}: Params) {
+  const { intervalString } = getBucketSize({
+    start: new Date(start).getTime(),
+    end: new Date(end).getTime(),
```

Comment:
Q: Isn't `start` and `end` already a timestamp?

Reply:
good catch, yes I updated here https://github.com/elastic/kibana/pull/189873/commits/c9601d96bd5177b255c81c5b7c2f7b6c85a00c89